### PR TITLE
Bluetooth: Controller: Restrict AD Data to BT_CTLR_ADV_DATA_LEN_MAX

### PR DESCRIPTION
**Pull request by cvinayak**
Strictly restrict AD Data length to BT_CTLR_ADV_DATA_LEN_MAX
when there can be free bytes in Advertising PDU with common
extended header format of less that the maximum 64 bytes.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

### CODE REVIEW

File: subsys/bluetooth/controller/ll_sw/ull_adv_aux.c [hunk 1]
```diff
@@ -316,8 +316,9 @@ uint8_t ll_adv_aux_sr_data_set(uint8_t handle, uint8_t op, uint8_t frag_pref,
 	/* Check if data will fit in remaining space */
 	/* TODO: need aux_chain_ind support */
 	ext_hdr_len = sr_dptr - &sr_com_hdr->ext_hdr_adv_data[0];
-	if ((PDU_AC_EXT_HEADER_SIZE_MIN + ext_hdr_len + len) >
-	    PDU_AC_PAYLOAD_SIZE_MAX) {
+	if ((len > CONFIG_BT_CTLR_ADV_DATA_LEN_MAX) ||
+	    ((PDU_AC_EXT_HEADER_SIZE_MIN + ext_hdr_len + len) >
+	     PDU_AC_PAYLOAD_SIZE_MAX)) {
```
Comment (Thalley) [inline subsys/bluetooth/controller/ll_sw/ull_adv_aux.c:321]: Hmm, you are changing 3 places that follows this pattern:

```c
if (len > MAX_A || (b + len) > MAX_B)
```

Does it make sense to add a macro or function for this check? 
If not, fine to keep as is

**Review by Thalley — DISMISSED** [review]

File: subsys/bluetooth/controller/ll_sw/ull_adv_aux.c (hunk 1 above)
Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/ull_adv_aux.c:321]: This code would change as part of long AD data support that is work in progress wherein total AD data length will have to be compared with `CONFIG_BT_CTLR_ADV_DATA_LEN_MAX` and the particular PDUs AD data fragment len with `PDU_AC_PAYLOAD_SIZE_MAX`. We can revisit use of macro then.

**Review by Thalley — DISMISSED** [review]
Did you verify if this breaks any existing samples?

**Comment by cvinayak** [comment]
> Did you verify if this breaks any existing samples?

As no samples change the default value for `CONFIG_BT_CTLR_SCAN_DATA_LEN_MAX` of 31, the change in the PR does not break any. I have tested iso_broadcast, iso_receive, periodic_adv and periodic_sync samples.

The bsim_bt_audio test has been updated accordingly as it had explicit `CONFIG_BT_CTLR_SCAN_DATA_LEN_MAX` value assigned.

**Comment by cvinayak** [comment]
@Thalley Updated so that Memory Capacity Exceeded error is returned as required by the LE Set Extended Advertising Data command and LE Set Periodic Advertising Data command.

**Review by Thalley — APPROVED** [review]

**Review by kruithofa — APPROVED** [review]

**Pull request merged**